src/js/main.js

```diff
@@ -1,3 +1,5 @@
+// This absolute unit controols my background image. I don't know a better way and I ened up chasing my tail for hours trying to make it better. Might showing me how the pro's do it?
```

Comment:
@jamesives

Reply:
This code is not complete, and there's some math that you'll need to figure out to get it to work properly which I've commented. It's pretty late here and my brain is a little bit cooked from the day, but I hope it's enough to point you in the right direction!

My intention with these refactors is to prevent you from writing a bunch of if statements for each background image, and instead neatly chop up the page into interval sections so once the user hits a certain point it will switch the background image programmatically. This means it should still work correctly no matter how much content you add to the page, and no matter how many background images you decide to add to the page. 

```javascript
window.addEventListener("scroll", function () {
  // When a user starts scrolling...
  const scrollPosition = Math.round(this.scrollY); // Grab the scroll position, rounding these suckers off because aint nobody got time for that
  const documentHeight = document.documentElement.scrollHeight; // Gonna need this for some fancy pants math
  const userViewingAt = (scrollPosition / documentHeight) * 100; // Where the user is currently scrolling
  const backgroundELement = document.querySelector("[data-target='portrait']"); // The body element for the background

  // Store each image in an array
  const backgroundImages = ['portrait_1', 'portrait_2', 'portrait_3', 'portrait_4', 'portrait_5', 'portrait_6', 'portrait_7', 'portrait_8', 'portrait_9']
  
  // Divide the document height by the length of the items in the array. These will be the intervals that this should transition at.
  // The math here is not correct, I can't recall the right scroll property to get the height relative to this.scrollY as it's late here.
  const transitionIntervals = documentHeight / backgroundImages.length;

  // Generic function to set the background image. 
  // Not really sure this abstraction is needed, but it prevents it on running on every scroll cycle.
  const setBackgroundImage = () => {
    backgroundELement.setAttribute("class", " ");
    backgroundELement.classList.add(backgroundImages[currentlyViewing]);
  }

  let currentlyViewing = 0;

  // If the user is viewing at a point which is greater than the transition interval relative to the currentlyViewing number,
  // then we incriment the number.

  // Otherwise we decrease the number, again the math equation isn't right here but it should be enough to get you started.
  // The goal is to not have this fire on every single scroll, and only at points where a transition should be made.
  // Each time one of these cycles fires it will set the background image to the index of what is currently being viewed.
  if (userViewingAt > transitionIntervals * currentlyViewing) {
    currentlyViewing = currentlyViewing + 1;

    setBackgroundImage()
  } else if (userViewingAt < transitionInterviews / currentlyViewing) {
    currentlyViewing = currentlyViewing - 1;

    setBackgroundImage();
  }
});
```

Let me know if that helps or if you have any more questions.